### PR TITLE
add swidth call, fixes #1051

### LINKED ISSUE #1051: cjk screen output formatting trouble
chinese, japanese, korean "monospace" terminal fonts use 2 cells per glyph, so using len(text) gives an incorrect value if text has cjk unicode code points.

this leads to wrong output in the `--progress` "single line output" - the amount of spaces to "fill" the line and erase previous output is computed wrongly.

there is https://pypi.python.org/pypi/wcwidth to compute the correct width.

3 ideas:
- just require it via setup.py
- bundle it in borg.support
- use a try / except ImportError and use a len(text) dummy if it can't be imported.

Screenshot: borg_jap

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
on posix platforms, this is a small wrapper around wcswidth(3).
on non-supported platforms, it just uses len(s), as before.

### CODE REVIEW

**Comment by enkore** [comment]
on windows the CRT doesn't seem to have that function, but I'm not sure whether the MSYS2 Python actually uses the CRT or a glibc port. If it's the CRT then we'd have to resort to Uniscribe I think, if one wanted to implement this for Windows as well.

**Comment by ThomasWaldmann** [comment]
How about adding a platform_posix and moving stuff there that is present on all posix platforms?
Then, in platform_{linux,freebsd,darwin}: from platform_posix import ...

**Comment by enkore** [comment]
Yeah

**Comment by ThomasWaldmann** [comment]
@enkore I'ld like to backport this to 1.0-maint so it gets into 1.0.3. But I think without the platform_posix / platform_base part - so I'ld just copy it to the places where needed.

What do you think?

**Comment by enkore** [comment]
Yes. Just need to be careful when back-merging it ;)

**Comment by ThomasWaldmann** [comment]
Circular import at helpers.py line 35 (from .platform import swidth)

**Comment by enkore** [comment]
Move the Statistics class to archive.py?

**Comment by ThomasWaldmann** [comment]
Yup, doing that right now.

**Comment by ThomasWaldmann** [comment]
Consider the amount of change to fix a cosmetic but, 1.0-maint just got less advisable i think.

**Comment by ThomasWaldmann** [comment]
merge as is?

File: borg/platform_linux.pyx [hunk 1]
```diff
@@ -5,6 +5,8 @@ import stat
 
 from .helpers import posix_acl_use_stored_uid_gid, user2uid, group2gid, safe_decode, safe_encode
 from .platform_base import SyncFile as BaseSyncFile
+from .platform_posix import swidth
+
```
Comment (enkore) [inline borg/platform_linux.pyx]: Do these in platform.py instead? Considering Windows would also only add ~two lines there.

Comment (ThomasWaldmann) [inline borg/platform_linux.pyx]: guess over time it'll become more. also, it avoids the temptation to just duplicate the code. :)

Comment (enkore) [inline borg/platform_linux.pyx]: fair enough

**Pull request merged**